metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py

```diff
@@ -326,14 +348,39 @@ def _get_bigquery_log_entries(
         # between query events and read events is complete. For example, this helps us
         # handle the case where the read happens within our time range but the query
         # completion event is delayed and happens after the configured end time.
+
+        # Can safely access the first index of the allow list as it by default contains ".*"
+        use_allow_filter = self.config.table_pattern and (
+            len(self.config.table_pattern.allow) > 1
```

Comment:
Shouldn't this be`>=1` here.  The more pythonic way to this check is simply  test for `self.config.table_pattern.allow` and eliminate the check on len.